src/arch/src/aarch64/regs.rs

```diff
@@ -196,6 +207,236 @@ pub fn read_mpidr(vcpu: &VcpuFd) -> Result<u64> {
     vcpu.get_one_reg(MPIDR_EL1).map_err(Error::GetSysRegister)
 }
 
+/// Get the state of the core registers.
+///
+/// # Arguments
+///
+/// * `vcpu` - Structure for the VCPU that holds the VCPU's fd.
+/// * `state` - Structure for returning the state of the core registers.
+pub fn save_core_registers(vcpu: &VcpuFd, state: &mut kvm_regs) -> Result<()> {
```

Comment:
It looks like we save the entire `kvm_regs` structure. Is it possible to also use `get_reg_list()` here ? instead of iterating through the entire structure manually ?

Reply:
Sorry, I'm not very familiar with ARM and this KVM methods. so `get_reg_list()` doesn't contain the core registers ?

Reply:
I am wondering too if we can get the core registers by using `get_reg_list()` and retain only what is not `system` registers. Would this filtering result in other registers besides core registers?

Reply:
OOh now I understand what you are saying. Indeed we can do it this way, nothing that can stop us. Never realized it since I started development with saving/retoring core_regs and then discovered reg list for the system registers. I updated the code to your suggestion.

Reply:
@iulianbarbu @serban300 I reverted this change as it was not working on the arm CI machines. 
The reason is that prior to this fix: https://patchwork.kernel.org/project/linux-arm-kernel/patch/20190620112301.138137-3-marc.zyngier@arm.com/, KVM_GET_REGLIST would also return invalid registers ids (not only core and system ids).
What tricked me into thinking that it worked was the fact that my dev machine has 5.3.0 kernel while on the CI machines the host kernel is version 4.14 and does not contain the fix mentioned above. Please re-review the reverted code!

Reply:
That's unfortunate :( Isn't there any way to filter these invalid registers from the userspace ?

Reply:
Yes, the code was super simple and nice before indeed. I spent quite some time trying to figure out what is happening, looked in kernel code for some way to filter those invalid registers and did not find any way. What happens is that KVM_GRET_REG_LIST returns a list of registers that apparently look like core registers but when I call kvm_get_one_reg the kernel returns EINVAL. Looking at the fix posted by Mark Zyngier regarding this issue, I see that he also uses `struct kvm_regs` for validating those, so it seems that this is the safest method to do it.